py/dml/ctree.py

```diff
@@ -187,6 +192,43 @@ def assert_comparable_types(site, expr1, expr2, equality):
         typ2 = realtype(expr2.ctype())
         raise EILLCOMP(site, expr1, typ1, expr2, typ2)
 
+class CoverityComplianceBlock:
+    active = False
+
+    def __init__(self, checker, classification):
+        self.checker = checker
+        self.classification = classification
+
+    def __enter__(self):
+        self.prev_context = CoverityComplianceBlock.active
+        CoverityComplianceBlock.active = True
+        if not self.prev_context:
+            out('#pragma GCC diagnostic push\n')
+            out('#pragma GCC diagnostic ignored "-Wunknown-pragmas"\n')
+        out('#pragma coverity compliance block '
+            + f'{self.classification} {self.checker}\n')
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        CoverityComplianceBlock.active = self.prev_context
+        if exc_type is None and not CoverityComplianceBlock.active:
+            out('#pragma coverity compliance end_block\n')
+            out('#pragma GCC diagnostic pop\n')
+
+def mkCoverityComplianceBlock(checker, classification='deviate'):
```

Comment:
The (arguably weird) mkStuff naming convention is currently used only for `Code` subclasses. Better to start with the more Pythonic name `make_coverity_compliance_block` and hillclimb naming from there.

Reply:
Removed